experimental/packages/opentelemetry-sdk-metrics-base/src/state/MeterProviderSharedState.ts

```diff
@@ -30,7 +30,7 @@ export class MeterProviderSharedState {
 
   metricCollectors: MetricCollector[] = [];
 
-  meters: Map<string, Meter> = new Map();
+  meters: Meter[] = [];
```

Comment:
This just to make the iteration in `MetricCollector.ts` easier?

Reply:
This removes the identity of meters in the registry, i.e. we don't need a key to register a meter.